controllers/front/ContactController.php

```diff
@@ -63,6 +63,14 @@ public function postProcess()
 
                 $id_order = (int)$this->getOrder();
 
+                /**
+                 * Check if customer select his order.
+                 */
+                if (!empty($id_order)) {
+                    $order = new Order($id_order);
+                    $id_order = (int) $order->id_customer === (int) $customer->id ? $id_order : 0;
```

Comment:
Be aware that this:
```php
(int) $order->id_customer === (int) $customer->id
```

Is basically equivalent to doing this:
```php
$order->id_customer == $customer->id
```

Reply:
Even if I'm agree with you, I prefer cast to integer because it's an id =)